External/Plugins/HaXeContext/Context.cs

```diff
@@ -903,65 +903,143 @@ public override bool IsImported(MemberModel member, int atLine)
+        /// <returns></returns>
+        public MemberList ResolveStaticExtensions(ClassModel target, FileModel inFile)
+        {
+            var result = new MemberList();
+            IEnumerable<MemberModel> imports = inFile.Imports.Items;
+            if (GetCurrentSDKVersion() >= "3.3.0") imports = ResolveDefaults(inFile.Package).Items.Concat(imports);
+            var importModels = imports.Where(it => it.Flags.HasFlag(FlagType.Using)).ToArray();
+            for (var i = importModels.Length - 1; i >= 0; i--)
+            {
+                var import = importModels[i];
+                var type = ResolveType(import.Name, inFile, false);
+                if (type.IsVoid() || type.Members.Count == 0) continue;
+                var access = TypesAffinity(target, type);
+                var extends = target;
+                while (!extends.IsVoid())
                 {
-                    if (import.Name == cname)
+                    foreach (MemberModel member in type.Members)
                     {
-                        if (import.Type.Length > import.Name.Length)
+                        if ((member.Access & access) > 0 && member.Flags.HasFlag(FlagType.Static | FlagType.Function) && member.Parameters?.Count > 0)
                         {
-                            var type = import.Type;
-                            int temp = type.IndexOf('<');
-                            if (temp > 0) type = type.Substring(0, temp);
-                            int dotIndex = type.LastIndexOf('.');
-                            if (dotIndex > 0) package = type.Substring(0, dotIndex);
+                            var extendsType = extends.Type;
+                            var firstParamType = member.Parameters[0].Type;
+                            var index = extendsType.IndexOf('<');
+                            if (index != -1) extendsType = extendsType.Remove(index);
+                            index = firstParamType.IndexOf('<');
+                            if (index != -1) firstParamType = firstParamType.Remove(index);
+                            if (firstParamType != extendsType) continue;
+                            var newMember = (MemberModel)member.Clone();
+                            newMember.Parameters.RemoveAt(0);
+                            newMember.Flags = FlagType.Dynamic | FlagType.Function;
+                            newMember.InFile = type.InFile;
+                            result.Add(newMember);
                         }
-                        found = true;
-                        break;
                     }
+                    extends.ResolveExtends();
+                    extends = extends.Extends;
                 }
-                if (!found && cname == "Function") return stubFunctionClass;
             }
-
-            return GetModel(package, cname, inPackage);
+            if (result.Count > 0)
+            {
+                result.Items.RemoveAll(extension =>
```

Comment:
Can you add some comments explaining what the different phases of this function do? This last one isn't clear in particular.